src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -210,6 +214,96 @@ function digest_boundary_conditions(boundary_conditions::AbstractArray, mesh, so
     throw(ArgumentError("Please use a (named) tuple instead of an (abstract) array to supply multiple boundary conditions (to improve performance)."))
 end
 
+function check_periodicity_mesh_boundary_conditions(mesh::TreeMesh{1},
+                                                    boundary_conditions)
+    @unpack x_neg, x_pos = boundary_conditions
+    if isperiodic(mesh.tree, 1) && (x_neg != BoundaryConditionPeriodic() ||
+        x_pos != BoundaryConditionPeriodic())
+        @warn "For periodic mesh non-periodic boundary conditions are supplied.
```

Comment:
Shouldn't we throw an error in this case, too?

Reply:
Please make it an error. The current implementation is buggy in allowing a non-sensible parameter set in the first place.

Reply:
I agree with this. This PR is a bug fix.

Reply:
Ok :+1: